ui/components/app/transaction-status-label/transaction-status-label.test.js

```diff
@@ -63,6 +79,152 @@ describe('TransactionStatusLabel Component', () => {
       <TransactionStatusLabel {...props} />,
+      store,
+    );
+
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should render statusText properly when is custodyStatusDisplayText is defined', () => {
+    const props = {
+      custodyStatusDisplayText: 'test',
+    };
+
+    const { getByText } = renderWithProvider(
+      <TransactionStatusLabel {...props} />,
+      store,
+    );
+
+    expect(getByText(props.custodyStatusDisplayText)).toBeVisible();
+  });
+
+  it('should display the correct status text and tooltip', () => {
+    const mockShortText = 'Short Text Test';
+    const mockLongText = 'Long Text Test';
+    const props = {
+      status: 'approved',
+      custodyStatus: 'approved',
+    };
+    const customMockStore = {
+      metamask: {
+        custodyStatusMaps: {
+          jupiter: {
+            approved: {
+              shortText: mockShortText,
+              longText: mockLongText,
+            },
+          },
+        },
+        selectedAddress: '0x0dcd5d886577d5081b0c52e242ef29e70be3e7bc',
+        identities: {
+          '0x0dcd5d886577d5081b0c52e242ef29e70be3e7bc': {
+            address: '0x0dcd5d886577d5081b0c52e242ef29e70be3e7bc',
+            name: 'Account 1',
+          },
+        },
+        keyrings: [
+          {
+            type: 'Custody - Jupiter',
+            accounts: ['0x0dcd5d886577d5081b0c52e242ef29e70be3e7bc'],
+          },
+        ],
+      },
+    };
+
+    store = createMockStore(customMockStore);
+
+    const { getByText, container } = renderWithProvider(
+      <TransactionStatusLabel {...props} />,
+      store,
+    );
+
+    expect(getByText(mockShortText)).toBeVisible();
+    expect(container).toMatchSnapshot();
+  });
+  it('should display the error message when there is an error', () => {
+    const mockShortText = 'Short Text Test';
+    const mockLongText = 'Long Text Test';
+    const props = {
+      status: 'approved',
+      custodyStatus: 'approved',
+      error: { message: 'An error occurred' },
+    };
+    const customMockStore = {
+      metamask: {
+        custodyStatusMaps: {
+          jupiter: {
+            approved: {
+              shortText: mockShortText,
+              longText: mockLongText,
+            },
+          },
+        },
+        selectedAddress: '0x0dcd5d886577d5081b0c52e242ef29e70be3e7bc',
+        identities: {
+          '0x0dcd5d886577d5081b0c52e242ef29e70be3e7bc': {
+            address: '0x0dcd5d886577d5081b0c52e242ef29e70be3e7bc',
+            name: 'Account 1',
+          },
+        },
+        keyrings: [
+          {
+            type: 'Custody - Jupiter',
+            accounts: ['0x0dcd5d886577d5081b0c52e242ef29e70be3e7bc'],
+          },
+        ],
+      },
+    };
+
+    store = createMockStore(customMockStore);
+
+    const { container } = renderWithProvider(
+      <TransactionStatusLabel {...props} />,
+      store,
+    );
+
+    expect(container).toMatchSnapshot();
```

Comment:
could you just inspect the error message?
`expect(dom_selection_here.textContent).toMatchInlineSnapshot)` :)